packages/client/ui/react-ui/src/components/auth/PasskeyPrompt.tsx

```diff
@@ -0,0 +1,211 @@
+import FingerprintIcon from "@/icons/fingerprint";
+import PasskeyIcon from "@/icons/passkey";
+import PasskeyPromptLogo from "@/icons/passkeyPromptLogo";
+import { Button } from "@headlessui/react";
+import React, { ReactNode } from "react";
+
+function PasskeyPromptCore({
+    title,
+    content,
+    primaryButton,
+    secondaryButton,
+}: {
+    title: string;
+    content: ReactNode;
+    primaryButton: ReactNode;
+    secondaryButton?: ReactNode;
+}) {
+    return (
+        <div
+            style={{
+                backgroundColor: "white",
+                maxWidth: "416px",
+                width: "100%",
+                borderRadius: "16px",
+                padding: "48px 32px",
+            }}
+        >
+            <div style={{ display: "flex", justifyContent: "center", left: "12px", position: "relative" }}>
+                <PasskeyPromptLogo />
+            </div>
+
+            <div style={{ display: "flex", justifyContent: "center" }}>
+                <p style={{ fontSize: "18px", fontWeight: "bold", color: "#20343E" }}>{title}</p>
+            </div>
+
+            <div style={{ marginTop: "24px", marginBottom: "36px" }}>{content}</div>
+
+            <div style={{ display: "flex", justifyContent: "center", flexDirection: "column", gap: "16px" }}>
+                {primaryButton}
+                {secondaryButton}
+            </div>
+        </div>
+    );
+}
+
+type PromptType = "create-wallet" | "transaction" | "error" | "not-supported";
+
+export function PasskeyPrompt({ type }: { type: PromptType }) {
+    // These components are currently assembled based on the mockups.
+    if (type === "create-wallet") {
+        return (
+            <PasskeyPromptCore
```

Comment:
For keeping the components simple and easy to change in the future i'm gonna keep them least abstracted as possible. I agree that abstracting these components could be nice but because i'll be moving fast i can always change them as needed later on rather than having large layers of abstraction to work through.

I like to follow the AHA principles: https://kentcdodds.com/blog/aha-programming